### PR TITLE
added vm-operator chart

### PR DESCRIPTION
**Pull request by k1rk**
Added VM Operator chart. 
For now just bare operator deployment, without any additional functionality

### CODE REVIEW

File: charts/victoria-metrics-operator/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      serviceAccountName: {{ template "vm-operator.serviceAccountName" . }}
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+            - name: WATCH_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
```
Comment (f41gh7) [inline charts/victoria-metrics-operator/templates/deployment.yaml:32]: single namespace is not supported :( only watch all
```yaml
        - name: WATCH_NAMESPACE
          value: ""
```

File: charts/victoria-metrics-operator/values.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  limits:
+    cpu: 120m
+    memory: 320Mi
+  requests:
+    cpu: 80m
+    memory: 120Mi
+
+nodeSelector: {}
+tolerations: []
+affinity: {}
```
Comment (f41gh7) [inline charts/victoria-metrics-operator/values.yaml:31]: may you also add env
```yaml
env:
  []
  # - name: VM_remoteWrite_basicAuth_password
  #   valueFrom:
  #     secretKeyRef:
  #       name: auth_secret
  #       key: password
```

File: charts/victoria-metrics-operator/values.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+logLevel: "info"
+rbac:
+  # Specifies whether the RBAC resources should be created
+  create: true
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: "her"
```
Comment (f41gh7) [inline charts/victoria-metrics-operator/values.yaml:20]: lets leave it empty

File: charts/victoria-metrics-operator/values.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
+# Default values for victoria-metrics.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+image:
+  repository: victoriametrics/operator
+  tag: v0.2.1
+  pullPolicy: IfNotPresent
+imagePullSecrets: []
+nameOverride: "vvvm"
```
Comment (f41gh7) [inline charts/victoria-metrics-operator/values.yaml:9]: lets leave it empty

File: charts/victoria-metrics-operator/templates/cluster_role.yaml [hunk 5]
```diff
@@ -0,0 +1,269 @@
+{{- if .Values.rbac.create -}}
```
Comment (f41gh7) [inline charts/victoria-metrics-operator/templates/cluster_role.yaml:1]: kindly ask you to add psp option:

https://github.com/VictoriaMetrics/helm-charts/blob/master/charts/victoria-metrics-agent/templates/podsecuritypolicy.yaml

File: charts/victoria-metrics-operator/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+            - name: POD_NAME
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.name
+            - name: OPERATOR_NAME
+              value: {{ .Chart.Name }}
```
Comment (f41gh7) [inline charts/victoria-metrics-operator/templates/deployment.yaml:38]: please add {{ with .Values.env }}

**Review by f41gh7 — CHANGES_REQUESTED** [review]
Thanks for contribution!

File: charts/victoria-metrics-operator/values.yaml (hunk 2 above)
Comment (k1rk) [reply charts/victoria-metrics-operator/values.yaml:31]: i was thinking about it. but do we need to provide any env variables?

File: charts/victoria-metrics-operator/templates/deployment.yaml (hunk 1 above)
Comment (k1rk) [reply charts/victoria-metrics-operator/templates/deployment.yaml:32]: good catch

File: charts/victoria-metrics-operator/values.yaml (hunk 3 above)
Comment (k1rk) [reply charts/victoria-metrics-operator/values.yaml:20]: ouch, sorry. was testing

File: charts/victoria-metrics-operator/values.yaml (hunk 4 above)
Comment (k1rk) [reply charts/victoria-metrics-operator/values.yaml:9]: same here

File: charts/victoria-metrics-operator/values.yaml (hunk 2 above)
Comment (k1rk) [reply charts/victoria-metrics-operator/values.yaml:31]: added

File: charts/victoria-metrics-operator/templates/cluster_role.yaml (hunk 5 above)
Comment (k1rk) [reply charts/victoria-metrics-operator/templates/cluster_role.yaml:1]: will add in separate PR

File: charts/victoria-metrics-operator/templates/deployment.yaml (hunk 6 above)
Comment (k1rk) [reply charts/victoria-metrics-operator/templates/deployment.yaml:40]: done

**Review by f41gh7 — APPROVED** [review]
LGTM

**Pull request merged**